src/overlays/actors/ovl_Toudai/ac_toudai.c

```diff
@@ -1,51 +1,287 @@
+    } else {
+        this->unk2C8 = this->unk2C8 + (8.0f * var_fv0);
+        this->unk2CC = this->unk2CC + var_fv0;
+    }
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB894_jp.s")
+    add_calc(&this->unk2D0, 255.0f, 0.1f, 100.0f, 1.0f);
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB8D8_jp.s")
+void aTOU_init(Toudai* this, Game_Play* game_play UNUSED) {
+    aTOU_setup_action(this, 1);
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB918_jp.s")
+void aTOU_wait(Toudai* this, Game_Play* game_play UNUSED) {
+    if ((common_data.time.now_sec >= 0xFD20) || (common_data.time.now_sec < 0x4650)) {
+        this->unk2BC = 1;
+        this->skeletonInfo.frameControl.speed = 1.0f;
+        aTOU_setup_action(this, 2);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB9C8_jp.s")
+void aTOU_lighting(Toudai* this, Game_Play* game_play UNUSED) {
+    if ((common_data.time.now_sec < 64800) && (common_data.time.now_sec >= 18000)) {
+        aTOU_setup_action(this, 3);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/aTOU_actor_init.s")
+void aTOU_lightout(Toudai* this, Game_Play* game_play UNUSED) {
+    if (this->unk1E8 == 0x33) {
+        this->unk2BC = 0;
+        this->skeletonInfo.frameControl.speed = 0.0f;
+        aTOU_setup_action(this, 1);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AABAB8_jp.s")
+static BaseAnimationR* ani[] = { (BaseAnimationR*)0x0607EE6C, (BaseAnimationR*)0x06080A74 };
+// cKF_ba_r_obj_s_toudai
+// cKF_ba_r_obj_w_toudai
+static ToudaiActionFunc process[] = { aTOU_init, aTOU_wait, aTOU_lighting, aTOU_lightout };
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AABAD8_jp.s")
+void aTOU_setup_action(Toudai* this, s32 processIndex) {
+    s32 type;
 
+    if (processIndex == 0) {
+        type = (common_data.time.season == WINTER);
+        cKF_SkeletonInfo_R_init(&this->skeletonInfo, this->skeletonInfo.skeleton, ani[type], 1.0f, 100.0f, 51.0f, 0.0f,
+                                0.0f, ANIMATION_REPEAT, NULL);
+    }
+
+    this->unk2A0 = process[processIndex];
+    this->unk2B4 = processIndex;
+}
+
+void aTOU_actor_move(Toudai* this, Game_Play* game_play) {
+    gSegments[6] = OS_K0_TO_PHYSICAL(common_data.structureClip->unk_AC(STRUCTURE_TOUDAI));
+    this->unk174 = cKF_SkeletonInfo_R_play(&this->skeletonInfo);
+    this->unk1E8 = this->skeletonInfo.frameControl.currentFrame;
+    this->unk2A0(this, game_play);
+    aTOU_color_ctrl(this);
+}
+
+void aTOU_actor_init(Actor* thisx, Game_Play* game_play) {
+    Toudai* this = THIS;
+
+    mFI_SetFG_common(61706, this->actor.home.pos, 0);
+    aTOU_actor_move(this, game_play);
+    this->actor.update = (ActorFunc)aTOU_actor_move;
```

Comment:
If this is an actor func then you should change the parameters to take an Actor* instead of a Toudai